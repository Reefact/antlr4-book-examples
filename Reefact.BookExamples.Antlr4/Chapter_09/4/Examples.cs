#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using NFluent;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_09._4 {

    [UseReporter(typeof(CustomReporter))]
    public class Examples {

        [Fact]
        public void valid_input() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("f(34);");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(stat (fcall f ( (expr 34) )) ;)");
        }

        [Fact]
        public void missing_closure() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("f((34);");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

        [Fact]
        public void too_much_parentheses() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("f((34)));");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

    }

}