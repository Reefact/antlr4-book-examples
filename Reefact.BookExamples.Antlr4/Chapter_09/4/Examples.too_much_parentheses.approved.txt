line 1:8 Too many parentheses.
(stat (fcall f ( (expr ( (expr 34) )) ) )) ;)